src/main/java/frc/robot/subsystems/Launcher.java

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkBase.ControlType;
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
+import edu.wpi.first.wpilibj.PneumaticsModuleType;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Constants.LaunchConstants.LaunchDistance;
+
+public class Launcher extends SubsystemBase {
+  /** Creates a new Launcher. */
+  private CANSparkMax upper =
+      new CANSparkMax(Constants.LaunchConstants.upperCANID, MotorType.kBrushless);
+
+  private CANSparkMax lower =
+      new CANSparkMax(Constants.LaunchConstants.lowerCANID, MotorType.kBrushless);
+
+  private boolean tuningPIDS = false;
+
+  private DoubleSolenoid angleSwitcher =
+      new DoubleSolenoid(
+          PneumaticsModuleType.CTREPCM,
+          Constants.LaunchConstants.angleSwitchForwardChannel,
+          Constants.LaunchConstants.angleSwitchReverseChannel);
+
+  public Launcher() {
+    setPIDsDefault();
+    showPIDs();
+  }
+
+  private void showPIDs() {
+    SmartDashboard.putNumber("Launch P", upper.getPIDController().getP());
+    SmartDashboard.putNumber("Launch I", upper.getPIDController().getI());
+    SmartDashboard.putNumber("Launch D", upper.getPIDController().getD());
+  }
+
+  private void setPIDsDefault() {
+    upper.getPIDController().setP(Constants.LaunchConstants.launcherP);
+    upper.getPIDController().setI(Constants.LaunchConstants.launcherI);
+    upper.getPIDController().setD(Constants.LaunchConstants.launcherD);
+
+    lower.getPIDController().setP(Constants.LaunchConstants.launcherP);
+    lower.getPIDController().setI(Constants.LaunchConstants.launcherI);
+    lower.getPIDController().setD(Constants.LaunchConstants.launcherD);
```

Comment:
this function is just:
```java
updatePIDs(Constants.LaunchConstants.launcherP, Constants.LaunchConstants.launcherI, Constants.LaunchConstants.launcherD)
```

Reply:
we addressed this in https://github.com/e-moser/Robot/pull/3 and i merged that nitoo your branch.